src/middleware.ts

```diff
@@ -21,6 +32,21 @@ export async function middleware(req: NextRequest) {
     return NextResponse.redirect(new URL("/auth/login", req.url));
   }
 
+  if (isOnboardPage) {
+    if (isOnboarded) {
+      return NextResponse.redirect(new URL("/sessions", req.url));
+    }
+    if (isMemberPage) {
+      if (!isOnboarded) {
+        return NextResponse.redirect(new URL("/onboard/name", req.url));
+      }
+    }
+  }
```

Comment:
since this is page specific I think its better not to handle it in the middleware.

On the actual member page you can have a check that
```
if (!!firstName || !!lastName) {
    redirect("onboard/name")
}
```